ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/StatementExecutorTest.java

```diff
@@ -449,6 +451,68 @@ public void shouldEnforceReferentialIntegrity() {
 
   }
 
+  @Test
+  public void shouldFailCreateAsSelectIfExceedActivePersistentQueriesLimit() {
```

Comment:
These tests are good, but if you're up for a challenge, see if you can get them to read better still, e.g.

```
@Test
  public void shouldFailCreateAsSelectIfExceedActivePersistentQueriesLimit() {
    // Given:
    givenConfigWith(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG, 2);
    givenActiveQueryCount(2);
    final Command csasCommand = givenCommand("CREATE STREAM user2pv AS select * from pageview;");
    final CommandId csasCommandId =  new CommandId(CommandId.Type.STREAM, "_CSASGen2", CommandId.Action.CREATE);

    // When:
    statementExecutor.handleStatement(csasCommand, csasCommandId, Optional.empty());

    // Then: 
    final CommandStatus commandStatus = getStatus(csasCommandId);
    assertThat(commandStatus.getStatus(), equalTo(CommandStatus.Status.ERROR));
    assertThat(commandStatus.getMessage(),
        containsString("limit on number of active, persistent queries"));
  }
```

Where:
- `givenConfigWith` just sets up a ksqlConfig field.
- `givenActiveQueryCount` updates a KsqlEngine mock to reflect the new active query count.
- `givenCommand` creates the command you need.
- `getStatus` gets the status from the `statementExecutor` and checks it is present.

Same for your other tests in here.

Reply:
Yeah, that one's a doozy!  

This is looking much better. I'll add a few follow up _suggestions_ below.